pkg/apis/core/validation/validation.go

```diff
@@ -4866,7 +4866,16 @@ var supportedServiceIPFamilyPolicy = sets.NewString(string(core.IPFamilyPolicySi
 
 // ValidateService tests if required fields/annotations of a Service are valid.
 func ValidateService(service *core.Service) field.ErrorList {
-	allErrs := ValidateObjectMeta(&service.ObjectMeta, true, ValidateServiceName, field.NewPath("metadata"))
+	metaPath := field.NewPath("metadata")
+	allErrs := ValidateObjectMeta(&service.ObjectMeta, true, ValidateServiceName, metaPath)
+
+	topologyHintsVal, topologyHintsSet := service.Annotations[core.DeprecatedAnnotationTopologyAwareHints]
+	topologyModeVal, topologyModeSet := service.Annotations[core.AnnotationTopologyMode]
+
+	if topologyModeSet && topologyHintsSet && topologyModeVal != topologyHintsVal {
+		message := fmt.Sprintf("may not set both %s and %s annotations concurrently unless they are set to the same value", core.DeprecatedAnnotationTopologyAwareHints, core.AnnotationTopologyMode)
```

Comment:
you want to use `metaPath.Child("annotations").Key(core.AnnotationTopologyMode)` and an error like Sprintf("must match `annotations[%]` when both are specified", core.DeprecatedAnnotationTopologyAwareHints)

Reply:
That's _much better_, thanks!